File: server/router/system/email_template.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type EmailTemplateRouter struct{}

func (r *EmailTemplateRouter) InitEmailTemplateRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	group := Router.Group("emailTemplate").Use(middleware.OperationRecord())
	groupWithoutRecord := Router.Group("emailTemplate")
	groupWithoutAuth := PublicRouter.Group("emailTemplate")
	{
		group.POST("create", emailTemplateApi.CreateEmailTemplate)
		group.PUT("update", emailTemplateApi.UpdateEmailTemplate)
		group.DELETE("delete", emailTemplateApi.DeleteEmailTemplate)
		group.DELETE("deleteByIds", emailTemplateApi.DeleteEmailTemplateByIds)
		group.POST("copy", emailTemplateApi.CopyEmailTemplate)
		group.POST("batchStatus", emailTemplateApi.BatchStatusEmailTemplate)
		group.POST("move", emailTemplateApi.MoveEmailTemplate)
		group.POST("preview", emailTemplateApi.PreviewEmailTemplate)
		group.POST("testSend", emailTemplateApi.TestSendEmailTemplate)
		group.POST("export", emailTemplateApi.ExportEmailTemplate)
		group.POST("import", emailTemplateApi.ImportEmailTemplate)
	}
	{
		groupWithoutRecord.GET("list", emailTemplateApi.GetEmailTemplateList)
		groupWithoutRecord.GET("detail", emailTemplateApi.GetEmailTemplateDetail)
	}
	{
		groupWithoutAuth.GET("downloadByToken", emailTemplateApi.DownloadEmailTemplateByToken)
	}
}

File: web/src/api/emailTemplate.js
import service from '@/utils/request'

export const getEmailTemplateList = (params) => {
  return service({
    url: '/emailTemplate/list',
    method: 'get',
    params
  })
}

export const getEmailTemplateDetail = (params) => {
  return service({
    url: '/emailTemplate/detail',
    method: 'get',
    params
  })
}

export const createEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/create',
    method: 'post',
    data
  })
}

export const updateEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/update',
    method: 'put',
    data
  })
}

export const deleteEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/delete',
    method: 'delete',
    data
  })
}

export const deleteEmailTemplateByIds = (data) => {
  return service({
    url: '/emailTemplate/deleteByIds',
    method: 'delete',
    data
  })
}

export const copyEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/copy',
    method: 'post',
    data
  })
}

export const batchStatusEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/batchStatus',
    method: 'post',
    data
  })
}

export const moveEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/move',
    method: 'post',
    data
  })
}

export const previewEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/preview',
    method: 'post',
    data
  })
}

export const testSendEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/testSend',
    method: 'post',
    data
  })
}

export const exportEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/export',
    method: 'post',
    data
  })
}

export const importEmailTemplate = (data) => {
  return service({
    url: '/emailTemplate/import',
    method: 'post',
    headers: { 'Content-Type': 'multipart/form-data' },
    data
  })
}
